### DISCUSSION TITLE
No PersonalBest for several drivers Bahrain Race

### DISCUSSION THREAD

**Discussion by jhughes3818** [theOehrly/Fast-F1#138] [General]
using .pickfastest() is returning NaT for several drivers in the race at Bahrain. Seems to be due to no lap having IsPersonalBest == True. It worked fine before the recent update.

Seems to be fine for Saudi Arabia, but haven't checked any other races.

**Comment by theOehrly** [comment]
I cannot reproduce that. The following code prints the fastest lap for every driver at the Bahrain Race. Not NaT values are printed, only valid lap times.

```
import fastf1

fastf1.Cache.enable_cache('tmp_cache')

session = fastf1.get_session(2022, 'Bahrain', 'R')
laps = session.load(telemetry=False, weather=False)

for driver in session.drivers:
    fastest = session.laps.pick_driver(driver).pick_fastest()
    print(driver, fastest['LapTime'])
```

Still, you could use `.pick_fastest(only_by_time=True)`, to get exactly the old behavior.

**Reply by jhughes3818** [reply]
Ah okay yeah, I just realised I was filtering for `IsAccurate == True`. The above code returns a lap for Leclerc for which IsAccurate is false
